src/pgduckdb_types.cpp

```diff
@@ -702,6 +850,31 @@ ConvertDecimal(const NumericVar &numeric) {
 	return (NumericIsNegative(numeric) ? -base_res : base_res);
 }
 
+int
+GetEnumPosition(Datum enum_oid) {
```

Comment:
I think for clarity it would be good to not use the name `enum_oid` at all in this PR, but use `enum_type_oid` and `enum_member_oid`. Right now you use `enum_oid` for both which is a bit confusing.